cmd/kubeadm/app/discovery/token/token.go

```diff
@@ -86,23 +86,9 @@ func RetrieveValidatedConfigInfo(cfg *kubeadmapi.JoinConfiguration) (*clientcmda
 			return true, nil
 		})
 
-		// Validate the MAC on the kubeconfig from the ConfigMap and load it
-		insecureKubeconfigString, ok := insecureClusterInfo.Data[bootstrapapi.KubeConfigKey]
-		if !ok || len(insecureKubeconfigString) == 0 {
-			return nil, errors.Errorf("there is no %s key in the %s ConfigMap. This API Server isn't set up for token bootstrapping, can't connect",
-				bootstrapapi.KubeConfigKey, bootstrapapi.ConfigMapClusterInfo)
-		}
-		detachedJWSToken, ok := insecureClusterInfo.Data[bootstrapapi.JWSSignatureKeyPrefix+token.ID]
-		if !ok || len(detachedJWSToken) == 0 {
-			return nil, errors.Errorf("token id %q is invalid for this cluster or it has expired. Use \"kubeadm token create\" on the control-plane node to create a new valid token", token.ID)
-		}
-		if !bootstrap.DetachedTokenIsValid(detachedJWSToken, insecureKubeconfigString, token.ID, token.Secret) {
-			return nil, errors.New("failed to verify JWS signature of received cluster info object, can't trust this API Server")
-		}
-		insecureKubeconfigBytes := []byte(insecureKubeconfigString)
-		insecureConfig, err := clientcmd.Load(insecureKubeconfigBytes)
+		insecureKubeconfigBytes, insecureConfig, err := getInsecureConfig(insecureClusterInfo, token)
```

Comment:
The idea of my last proposition was to have a call like:
```
... := getInsecureConfig(insecureClient, token)
```

and have the `PollImmediateInfinite` stuff from above should be shared between `getSecureKubeconfig` and `getInsecureKubeconfig` (be in a third func).

Reply:
I see, I just think it's a little small just PollImmediateInfinite as a func. While it‘s the better to share it between getSecureKubeconfig and getInsecureKubeconfig. I've revised it. Thanks!

Reply:
@rosti hi, how about this pr? Is there anything else to be modified?